tests/rbd/test_rbd_immutable_cache.py

```diff
@@ -0,0 +1,225 @@
+        status_output = execute_command(client_node, status_command)
+
+        # Verify object cache daemon got activated successfully
+        if "Active" not in status_output:
+            log.debug(status_output)
+            log.error(f"{service_name} not activated")
+            return 1
+        log.info("ceph-immutable-object-cache daemon got activated successfully")
+
+    except Exception as err:
+        log.error(err)
+        return 1
+
+    return 0
+
+
+def verify_cache_objects(rbd, node, **kw):
+    """
+    Method to validate cache objects after configuring immutable cache daemon.
+
+    Args:
+        rbd: RBD object
+        node: cache client node
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+    pool = config["rep_pool_config"]["pool"]
+    image = config["rep_pool_config"]["image1"]
+    image_size = config["rep_pool_config"]["size"]
+
+    snap_name = f"{pool}/{image}@snap1"
+    clone = "cloned_image1"
+    try:
+        rbd.create_image(
+            pool,
+            image,
+            size=image_size,
+        )
+
+        run_fio(image_name=image, pool_name=pool, client_node=node)
+
+        rbd.snap_create(pool, image, "snap1")
+
+        rbd.protect_snapshot(snap_name)
+
+        rbd.create_clone(
+            snap_name,
+            pool,
+            clone,
+        )
+
+        rbd.flatten_clone(pool, clone)
+
+        # Read IO from cloned images to get cache files
+        run_fio(image_name=clone, pool_name=pool, client_node=node, io_type="read")
+
+        # waiting for cache files getting generated
+        time.sleep(60)
+        import pdb
+
+        pdb.set_trace()
+        execute_command(node, f"cd {immutable_cache_path}")
+        execute_command(node, "ls >> cache_files")
+
+        file = node.remote_file(
+            sudo=True, file_name=f"{immutable_cache_path}/cache_files", file_mode="r"
+        )
+        content = file.read()
+        if not content:
+            log.error("No cache files found.")
+            return 1
+        else:
+            log.info(f"Cache files found: {content}")
+            return 0
+
+    except Exception as err:
+        log.error(err)
+        return 1
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to configure and validate ceph immutable object cache files.
+
+    Args:
+        ceph_cluster: ceph cluster object
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    log.info(
+        "Starting CEPH-83575581 - Test to configure and validate"
+        "ceph immutable object cache files."
+    )
+
+    config = kw.get("config")
+    rbd_obj = initial_rbd_config(**kw)["rbd_reppool"]
+    client_node = get_node_by_id(ceph_cluster, config["client"])
+
+    # To configure immutable object cache
+    # if configure_immutable_cache(rbd_obj, client_node, "user3"):
```

Comment:
Why are these lines commented?

Reply:
uncommented i was testing some changes at that time sorry.